File: generated-sources/kotlin-spring/mojang-api/src/main/kotlin/com/github/asyncmc/mojang/api/kotlin/spring/model/SecurityChallenge.kt
package com.github.asyncmc.mojang.api.kotlin.spring.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import com.github.asyncmc.mojang.api.kotlin.spring.model.SecurityAnswerId
import com.github.asyncmc.mojang.api.kotlin.spring.model.SecurityQuestion
import javax.validation.Valid
import javax.validation.constraints.*

/**
 * Binds a question to an answer id, the answer must respect the answer id.
 * @param question 
 * @param answer 
 */
data class SecurityChallenge (

        @get:NotNull 
        @JsonProperty("question") val question: SecurityQuestion,

        @get:NotNull 
        @JsonProperty("answer") val answer: SecurityAnswerId
) {

}

File: generated-sources/scala-gatling/mojang-api/src/gatling/scala/com/github/asyncmc/mojang/api/scala/gatling/model/SecurityChallenge.scala

package com.github.asyncmc.mojang.api.scala.gatling.model


case class SecurityChallenge (
    _question: SecurityQuestion,
    _answer: SecurityAnswerId
)
object SecurityChallenge {
    def toStringBody(var_question: Object, var_answer: Object) =
        s"""
        | {
        | "question":$var_question,"answer":$var_answer
        | }
        """.stripMargin
}
